### ISSUE TITLE
[Gallery][All Platform] `DataGrid` - The newly selected date from the calendar does not update the 'Date Planted' display box for that row.

### ISSUE THREAD

**Issue by sakshi173** [unoplatform/uno#16065]
### Current behavior

https://github.com/unoplatform/uno/assets/74563602/fec96829-5726-471b-90e4-ece1dca424df

### Expected behavior

The changed date should display.

### How to reproduce it (as minimally and precisely as possible)

1. Launch the Uno.Gallery app for 4X or 5X version for Android.
2. Click on “DataGrid” from the "Community Toolkit" list.
3. Go to 'Date Planted' section
4. Click on any existing date.
5. Calendar should open.
6. In the open calendar choose any date.
7. Now click on any other part of the screen.
8. Notice that the newly selected date from the calendar does not update the 'Date Planted' display box for that row.

### Works on UWP/WinUI

None

### Affected platforms

WebAssembly, Android, iOS, Skia (GTK on Linux/macOS/Windows)

**Comment by xperiandri** [comment]
I wonder what is this DataGrid from. Probably from Toolkit v7 as v8 does not contain a DataGrid

**Comment by morning4coffe-dev** [comment]
@xperiandri Yes, this one is still from the version 7.1.2. As you said the DataGrid is currently not supported on v8+, but I think the team is currently working on a simplified DataTable - https://github.com/CommunityToolkit/Labs-Windows/discussions/415

**Comment by jeromelaban** [comment]
@sakshi173 is the calendar showing up on the Windows version? It's very likely that it is not the case.

**Comment by sakshi173** [comment]
@jeromelaban, this issue is for canary build in chrome browser for WASM and also for Android and iOS  platform too, I am not sure what exactly is your question?

**Comment by jeromelaban** [comment]
@sakshi173 is the behavior you're expecting found in the windows version of the gallery?

**Comment by sakshi173** [comment]
@jeromelaban, okay now understood, let me check what's the functionality in Windows version.

**Comment by sakshi173** [comment]
@jeromelaban , I tried to check the window version behavior for the this defect but windows version is getting crash when click on the date box. Below is the video for windows version.

https://github.com/unoplatform/uno/assets/74563602/44a33fc7-4d44-44d5-9d5c-1b4707c782e3

**Comment by Youssef1313** [comment]
@jeromelaban Could it be related to the layout cycle noted in https://github.com/unoplatform/uno/issues/16064 ?

i.e, our current behavior is inline with WinUI that both crashes (but when we crash we gracefully handle the exception and don't really crash the whole app)?

**Comment by jeromelaban** [comment]
The fact that the windows version crashes indicates that the original code using the DataGrid is probably incorrect (either from the DataGrid itself, or from our integration from the gallery).
